buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -405,4 +465,80 @@ private static String destructiveDistroTestTaskName(ElasticsearchDistribution di
             distro.getFlavor(),
             distro.getBundledJdk());
     }
+
+    static Map<String, String> parseOsRelease(final List<String> osReleaseLines) {
+        final Map<String, String> values = new HashMap<>();
+
+        for (String line : osReleaseLines) {
+            final String trimmed = line.trim();
+
+            // ignore empty and comment lines
+            if (trimmed.isEmpty() || trimmed.startsWith("#")) {
+                continue;
+            }
+
+            final String[] parts = line.split("=", 2);
+            final String key = parts[0];
+            // remove optional leading and trailing quotes and whitespace
+            final String value = parts[1].replaceAll("^['\"]?\\s*", "").replaceAll("\\s*['\"]?$", "");
+            values.put(key, value);
+        }
+
+        return values;
+    }
+
+    static String deriveId(final Map<String, String> osRelease) {
+        return osRelease.get("ID") + "-" + osRelease.get("VERSION_ID");
+    }
+
+    /**
+     * The {@link DistroTestPlugin} generates a number of test tasks, some
+     * of which are Docker packaging tests. When running on the host OS
+     * i.e. not in a VM, only certain operating systems are supported. This
+     * method determines whether the Docker tests should be run on the host
+     * OS.
+     */
+    private static boolean shouldRunDockerTests() {
+        switch (OS.current()) {
+            case WINDOWS:
+                // Not currently supported.
+                logger.debug("shouldRunDockerTests = false on " + OS.WINDOWS);
+                return false;
+
+            case MAC:
+                // Assume that Docker for Mac is installed
```

Comment:
Do we _need_ to assume? Couldn't we leverage `DockerAvailability` here to make this a bit more robust? Also, should we even explicitly handle MAC since it's not an OS that we run _any_ packaging tests on.